src/web/lib/contributions.test.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    getLastOneOffContributionDate,
+    ONE_OFF_CONTRIBUTION_DATE_COOKIE,
+    SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE,
+} from './contributions';
+
+const setCookie = (name: string, value: string) => {
+    Object.defineProperty(window.document, 'cookie', {
+        writable: true,
+        value: `${name}=${value}`,
+    });
+};
+
+const clearAllCookies = () => {
+    const cookies = document.cookie.split(';');
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < cookies.length; i++) {
+        const cookie = cookies[i];
+        const eqPos = cookie.indexOf('=');
+        const name = eqPos > -1 ? cookie.substr(0, eqPos) : cookie;
+        document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT`;
+    }
+};
+
+describe('getLastOneOffContributionDate', () => {
+    beforeEach(clearAllCookies);
+
+    it('returns date from attributes cookie if only cookie found', () => {
+        const somePastDate = '2020-01-28';
+        const formattedPastDate = Date.parse(somePastDate).toString();
```

Comment:
What about explicitly writing out the formatted date, rather than using Date.parse...? It might be clearer to the reader what is expected.

Reply:
This cookie needs to be written out as `YYYY-MM-DD` but the function we're testing here will parse that into a timestamp, so I'm declaring both values here so we can save with the right format and test the value we get back from the function we're testing. I've renamed the vars to make this a bit more clear.